src/Functions/IFunction.h

```diff
@@ -155,12 +155,11 @@ class IFunctionBase
       */
     virtual bool isSuitableForConstantFolding() const { return true; }
 
-    /** Some functions like ignore(...) or toTypeName(...) always return constant result which doesn't depend on arguments.
-      * In this case we can calculate result and assume that it's constant in stream header.
-      * There is no need to implement function if it has zero arguments.
-      * Must return ColumnConst with single row or nullptr.
+    /** If function isSuitableForConstantFolding than, this method will be called during query analyzis
+      * if some arguments are constants. For example logical functions (AndFunction, OrFunction) can
+      * return they result based on some constant arguments.
       */
-    virtual ColumnPtr getResultIfAlwaysReturnsConstantAndHasArguments(const ColumnsWithTypeAndName & /*columns*/) const { return nullptr; }
+    virtual ColumnPtr getConstantResultForNonConstArguments(const ColumnsWithTypeAndName & /* arguments */) const { return nullptr; }
```

Comment:
I think we should add that `defaultImplementationForSomething` is ignored here.

Reply:
Fixed.